src/arrayMethodSort.js

```diff
@@ -4,8 +4,29 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length; i++) {
+      for (let k = 0; k < this.length; k++) {
+        const comparedEl = this[i];
+        const currentEl = this[k];
+
+        if (compareFunction === undefined) {
+          if (this[i].toString() < this[k].toString()) {
+            this[i] = currentEl;
+            this[k] = comparedEl;
+          }
+
+          continue;
+        }
+
+        if (compareFunction(this[i], this[k]) < 0) {
+          this[i] = currentEl;
+          this[k] = comparedEl;
+        }
+      }
+    }
+
+    return this;
```

Comment:
Consider adding a check to handle the case where the array is already sorted. This can help improve the performance of your sort function by avoiding unnecessary iterations.

Reply:
Do i really need it?